File: blobs/howto/dotnet/dotnet-v12/Authorize_DataLake.cs
using Azure;
using Azure.Core;
using Azure.Identity;
using Azure.Storage;
using Azure.Storage.Files.DataLake;
using System;
using System.Collections.Generic;
using System.Text;

namespace dotnet_v12
{
    public static class Authorize_DataLake
    {
        //-------------------------------------------------------------
        // Connect to the storage account - get Data Lake service client
        //----------------------------------------------------------

        // <Snippet_AuthorizeWithKey>
        public static DataLakeServiceClient GetDataLakeServiceClient(string accountName, string accountKey)
        {
            StorageSharedKeyCredential sharedKeyCredential =
                new StorageSharedKeyCredential(accountName, accountKey);

            string dfsUri = $"https://{accountName}.dfs.core.windows.net";

            DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClient(
                new Uri(dfsUri),
                sharedKeyCredential);

            return dataLakeServiceClient;
        }
        // </Snippet_AuthorizeWithKey>

        // ---------------------------------------------------------
        // Connect to the storage account (Azure AD - get Data Lake service client)
        //----------------------------------------------------------

        // <Snippet_AuthorizeWithAAD>
        public static DataLakeServiceClient GetDataLakeServiceClient(string accountName)
        {
            string dfsUri = $"https://{accountName}.dfs.core.windows.net";

            DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClient(
                new Uri(dfsUri),
                new DefaultAzureCredential());

            return dataLakeServiceClient;
        }
        // </Snippet_AuthorizeWithAAD>

        // <Snippet_AuthorizeWithSAS>
        public static DataLakeServiceClient GetDataLakeServiceClientSAS(string accountName, string sasToken)
        {
            string dfsUri = $"https://{accountName}.dfs.core.windows.net";

            DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClient(
                new Uri(dfsUri),
                new AzureSasCredential(sasToken));

            return dataLakeServiceClient;
        }
        // </Snippet_AuthorizeWithSAS>
    }
}

File: blobs/howto/Java/Java-v12/src/main/java/com/datalake/manage/Authorize_DataLake.java
package com.datalake.manage;

import com.azure.identity.*;
import com.azure.storage.common.StorageSharedKeyCredential;
import com.azure.storage.file.datalake.*;

public class Authorize_DataLake {
    
    // ---------------------------------------------------------
    // Connect to the storage account - use account key
    // ----------------------------------------------------------
    
    //<Snippet_AuthorizeWithKey>
    static public DataLakeServiceClient GetDataLakeServiceClient
    (String accountName, String accountKey){
        StorageSharedKeyCredential sharedKeyCredential =
            new StorageSharedKeyCredential(accountName, accountKey);

        DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClientBuilder()
            .endpoint("https://" + accountName + ".dfs.core.windows.net")
            .credential(sharedKeyCredential)
            .buildClient();

        return dataLakeServiceClient;
    }  
    //</Snippet_AuthorizeWithKey>

    // ---------------------------------------------------------
    // Connect to the storage account - use SAS
    // ----------------------------------------------------------
    
    //<Snippet_AuthorizeWithSAS>
    static public DataLakeServiceClient GetDataLakeServiceClientSAS(String accountName, String sasToken){
        DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClientBuilder()
            .endpoint("https://" + accountName + ".dfs.core.windows.net")
            .sasToken(sasToken)
            .buildClient();

        return dataLakeServiceClient;
    }  
    //</Snippet_AuthorizeWithSAS>
    
    // ---------------------------------------------------------
    // Connect to the storage account - use service principal and Azure AD
    // ----------------------------------------------------------
    
    //<Snippet_AuthorizeWithAzureAD>
    static public DataLakeServiceClient GetDataLakeServiceClient(String accountName){
        DefaultAzureCredential defaultCredential = new DefaultAzureCredentialBuilder().build();

        DataLakeServiceClient dataLakeServiceClient = new DataLakeServiceClientBuilder()
            .endpoint("https://" + accountName + ".dfs.core.windows.net")
            .credential(defaultCredential)
            .buildClient();

        return dataLakeServiceClient;
    } 
    //</Snippet_AuthorizeWithAzureAD> 
    
}
